src/components/ThemedToolbar/ThemedToolbar.tsx

```diff
@@ -1,12 +1,37 @@
 import React from "react";
-import { IonToolbar } from "@ionic/react";
+import { IonIcon, IonProgressBar, IonToolbar } from "@ionic/react";
+import { cloudOfflineOutline } from "ionicons/icons";
+import { useNetworkStatus } from "../../NetworkStatus";
 
 import classes from "./ThemedToolbar.module.css";
+import { useIsFetching, useIsMutating } from "@tanstack/react-query";
 
 interface ThemedToolbarProps extends React.ComponentProps<typeof IonToolbar> {}
 
 const ThemedToolbar: React.FC<ThemedToolbarProps> = (props) => {
-  return <IonToolbar {...props} className={classes.themedBackground} />;
+  const { isOnline } = useNetworkStatus();
+  const { children, ...rest } = props;
+  const isFetching = useIsFetching();
+  const isMutating = useIsMutating({ predicate: (m) => !m.state.isPaused });
+
+  return (
+    <>
+      {!isOnline && (
+        <IonToolbar color="medium" className={classes.offlineToolbar}>
+          <div className={classes.offlineWarning}>
+            <IonIcon icon={cloudOfflineOutline} />
+            <div>You are offline</div>
+          </div>
+        </IonToolbar>
+      )}
+      <IonToolbar {...rest} className={classes.themedBackground}>
+        {children}
+      </IonToolbar>
+      {(isFetching > 0 || isMutating > 0) && (
```

Comment:
I was surprised to see these values being compared with a number (i.e. 0), as—based upon their names—I had assumed they were boolean values.

Reply:
Me too! Before I added the `> 0` I was confused why the number 0 was appearing in the UI.